commit 3fbb32963323b34689f81753cb4ee4aa7be25ab3 (refs/changes/99/3099/2)
Author: Trent Mick <trentm@gmail.com>
Date:   2017-12-12T12:07:47-08:00 (1 year, 10 months ago)
    
    TRITON-27 vm-agent: update eslint to 2.x
    Reviewed by: Josh Wilsdon <josh@wilsdon.ca>
    Approved by: Josh Wilsdon <josh@wilsdon.ca>

diff --git a/.eslintrc b/.eslintrc
index ca453c3..c80f0bb 100644
--- a/.eslintrc
+++ b/.eslintrc
@@ -1,121 +1,119 @@
 {
-    // 0 = off, 1 = warn, 2 = err
-    "rules": {
-        "array-bracket-spacing": [2, "never"],
-        "block-scoped-var": 2,
-        "brace-style": 2,
-        "callback-return": [2, ["callback", "cb", "next"]],
-        "comma-dangle": [2, "never"],
-        "comma-spacing": [2, {"before": false, "after": true}],
-        "comma-style": [2, "last"],
-        "complexity": [1, 6],
-        "computed-property-spacing": [2, "never"],
-        "consistent-return": 2,
-        "consistent-this": [2, "self"],
-        "curly": [2, "all"],
-        "default-case": 2,
-        "dot-location": [2, "property"],
-        "dot-notation": [2, {"allowKeywords": true, "allowPattern": ""}],
-        "eqeqeq": [2, "smart"],
-        "eol-last": 2,
-        "func-names": 2,
-        "func-style": [2, "declaration"],
-        "global-require": 2,
-        "guard-for-in": 2,
-        "handle-callback-err": [2, "^.*(e|E)rr" ],
-        "indent": [2, 4, {"SwitchCase": 1}],
-        "key-spacing": [2, {"beforeColon": false, "afterColon": true}],
-        "linebreak-style": [2, "unix"],
-        "lines-around-comment": [2, { "beforeBlockComment": true, "allowBlockStart": true }],
-        "max-depth": [2, 5],
-        "max-nested-callbacks": [2, 5],
-        "max-len": [2, 80, 8, {"ignoreUrls": true}],
-        "new-parens": 2,
-        "newline-after-var": [2, "always"],
-        "no-array-constructor": 2,
-        "no-case-declarations": 2,
-        "no-cond-assign": [2, "always"],
-        "no-console": 1,
-        "no-constant-condition": 2,
-        "no-dupe-args": 2,
-        "no-dupe-keys": 2,
-        "no-duplicate-case": 2,
-        "no-else-return": 2,
-        "no-empty": 2,
-        "no-empty-label": 2,
-        "no-empty-pattern": 2,
-        "no-eq-null": 2,
-        "no-eval": 2,
-        "no-extend-native":2,
-        "no-extra-bind": 2,
-        "no-extra-semi": 2,
-        "no-floating-decimal": 2,
-        "no-implicit-coercion": 2,
-        "no-inner-declarations": 2,
-        "no-irregular-whitespace": 2,
-        "no-labels": 2,
-        "no-label-var": 2,
-        "no-loop-func": 2,
-        "no-lone-blocks": 2,
-        "no-lonely-if": 2,
-        "no-magic-numbers": [2, {ignore: [-1, 0, 1]}],
-        "no-mixed-requires": 2,
-        "no-mixed-spaces-and-tabs": 2,
-        "no-multi-spaces": [2, { exceptions: { "Property": false } }],
-        "no-multiple-empty-lines": [2, {"max": 2}],
-        "no-multi-str": 2,
-        "no-negated-condition": 2,
-        "no-nested-ternary": 2,
-        "no-new": 2,
-        "no-new-func": 2,
-        "no-new-object": 2,
-        "no-new-require": 2,
-        "no-new-wrappers": 2,
-        "no-param-reassign": 2,
-        "no-proto": 2,
-        "no-return-assign": [2, "always"],
-        "no-self-compare": 2,
-        "no-sequences": 2,
-        "no-shadow": 2,
-        "no-shadow-restricted-names": 2,
-        "no-spaced-func": 2,
-        "no-throw-literal": 2,
-        "no-trailing-spaces": 2,
-        "no-undef-init": 2,
-        "no-undefined": 2,
-        "no-unneeded-ternary": 2,
-        "no-unused-expressions": 2,
-        "no-unused-vars": 2,
-        "no-use-before-define": 2,
-        "no-useless-call": 2,
-        "no-useless-concat": 2,
-        "no-warning-comments": [1, { "terms": ["todo", "fixme", "xxx"], "location": "start" }],
-        "no-with": 2,
-        "object-curly-spacing": [2, "never"],
-        "one-var": [2, "never"],
-        "operator-linebreak": [2, "before"],
-        "padded-blocks": [2, "never"],
-        "radix": 2,
-        "quote-props": [2, "as-needed"],
-        "quotes": [2, "single"],
-        "semi": [2, "always"],
-        "semi-spacing": 2,
-        "sort-vars": [2, {"ignoreCase": true}],
-        "space-after-keywords": 2,
-        "space-before-blocks": 2,
-        "space-before-function-paren": [2, { "anonymous": "always", "named": "never" }],
-        "space-before-keywords": [2, "always"],
-        "space-in-parens": [2, "never"],
-        "space-infix-ops": 2,
-        "space-return-throw-case": 2,
-        "space-unary-ops": 2,
-        "spaced-comment": [2, "always"],
-        "vars-on-top": 2,
-        "wrap-iife": [2, "inside"],
-        "wrap-regex": 2
-    },
     "env": {
         "node": true
     },
-    "extends": "eslint:recommended"
+    "extends": [
+        "eslint:recommended"
+    ],
+    "rules": {
+        "array-bracket-spacing": ["error", "never"],
+        "block-scoped-var": "error",
+        "brace-style": "error",
+        "callback-return": ["error", ["callback", "cb", "next"]],
+        "comma-dangle": ["error", "never"],
+        "comma-spacing": ["error", {"before": false, "after": true}],
+        "comma-style": ["error", "last"],
+        "complexity": ["warn", 6],
+        "computed-property-spacing": ["error", "never"],
+        "consistent-return": "error",
+        "consistent-this": ["error", "self"],
+        "curly": ["error", "all"],
+        "default-case": "error",
+        "dot-location": ["error", "property"],
+        "dot-notation": ["error", {"allowKeywords": true, "allowPattern": ""}],
+        "eqeqeq": ["error", "smart"],
+        "eol-last": "error",
+        "func-names": "error",
+        "func-style": ["error", "declaration"],
+        "global-require": "error",
+        "guard-for-in": "error",
+        "handle-callback-err": ["error", "^.*(e|E)rr" ],
+        "indent": ["error", 4, {"SwitchCase": 1}],
+        "key-spacing": ["error", {"beforeColon": false, "afterColon": true}],
+        "linebreak-style": ["error", "unix"],
+        "lines-around-comment": ["error", { "beforeBlockComment": true, "allowBlockStart": true }],
+        "max-depth": ["error", 5],
+        "max-nested-callbacks": ["error", 5],
+        "max-len": ["error", 80, 8, {"ignoreUrls": true}],
+        "new-parens": "error",
+        "newline-after-var": ["error", "always"],
+        "no-array-constructor": "error",
+        "no-case-declarations": "error",
+        "no-cond-assign": ["error", "always"],
+        "no-console": "warn",
+        "no-constant-condition": "error",
+        "no-dupe-args": "error",
+        "no-dupe-keys": "error",
+        "no-duplicate-case": "error",
+        "no-else-return": "error",
+        "no-empty": "error",
+        "no-empty-pattern": "error",
+        "no-eq-null": "error",
+        "no-eval": "error",
+        "no-extend-native":"error",
+        "no-extra-bind": "error",
+        "no-extra-semi": "error",
+        "no-floating-decimal": "error",
+        "no-implicit-coercion": "error",
+        "no-inner-declarations": "error",
+        "no-irregular-whitespace": "error",
+        "no-labels": "error",
+        "no-label-var": "error",
+        "no-loop-func": "error",
+        "no-lone-blocks": "error",
+        "no-lonely-if": "error",
+        "no-magic-numbers": ["error", {ignore: [-1, 0, 1, 2]}],
+        "no-mixed-requires": "error",
+        "no-mixed-spaces-and-tabs": "error",
+        "no-multi-spaces": ["error", { exceptions: { "Property": false } }],
+        "no-multiple-empty-lines": ["error", {"max": 2}],
+        "no-multi-str": "error",
+        "no-negated-condition": "error",
+        "no-nested-ternary": "error",
+        "no-new": "error",
+        "no-new-func": "error",
+        "no-new-object": "error",
+        "no-new-require": "error",
+        "no-new-wrappers": "error",
+        "no-param-reassign": "error",
+        "no-proto": "error",
+        "no-return-assign": ["error", "always"],
+        "no-self-compare": "error",
+        "no-sequences": "error",
+        "no-shadow": "error",
+        "no-shadow-restricted-names": "error",
+        "no-spaced-func": "error",
+        "no-throw-literal": "error",
+        "no-trailing-spaces": "error",
+        "no-undef-init": "error",
+        "no-undefined": "error",
+        "no-unneeded-ternary": "error",
+        "no-unused-expressions": "error",
+        "no-unused-vars": "error",
+        "no-use-before-define": "error",
+        "no-useless-call": "error",
+        "no-useless-concat": "error",
+        "no-warning-comments": ["warn", { "terms": ["todo", "fixme", "xxx"], "location": "start" }],
+        "no-with": "error",
+        "object-curly-spacing": ["error", "never"],
+        "one-var": ["error", "never"],
+        "operator-linebreak": ["error", "before"],
+        "padded-blocks": ["error", "never"],
+        "radix": "error",
+        "quote-props": ["error", "as-needed"],
+        "quotes": ["error", "single"],
+        "semi": ["error", "always"],
+        "semi-spacing": "error",
+        "sort-vars": ["error", {"ignoreCase": true}],
+        "keyword-spacing": "error",
+        "space-before-blocks": "error",
+        "space-before-function-paren": ["error", { "anonymous": "always", "named": "never" }],
+        "space-in-parens": ["error", "never"],
+        "space-infix-ops": "error",
+        "space-unary-ops": "error",
+        "spaced-comment": ["error", "always"],
+        "vars-on-top": "error",
+        "wrap-iife": ["error", "inside"],
+        "wrap-regex": "error"
+    }
 }
diff --git a/package.json b/package.json
index a0f3171..8cd457f 100644
--- a/package.json
+++ b/package.json
@@ -9,7 +9,7 @@
         "bunyan": "1.5.1",
         "cueball": "2.1.1",
         "deep-diff": "0.3.3",
-        "eslint": "1.10.1",
+        "eslint": "2.13.1",
         "kthxbai": "0.4.0",
         "lstream": "0.0.4",
         "mockery": "1.4.0",
